java/code/src/com/suse/manager/reactor/messaging/RegistrationUtils.java

```diff
@@ -253,28 +255,25 @@ public static void subscribeMinionToChannels(SystemQuery systemQuery, MinionServ
                         // ActivationKey without base channel (SUSE Manager Default)
                         () -> {
                             Set<SUSEProduct> suseProducts = identifyProduct(systemQuery, server, grains);
-                            return findChannelsForProducts(suseProducts, minionId);
+                            Set<Channel> channelsForProducts = findChannelsForProducts(suseProducts, minionId);
+                            Optional<Channel> baseChannel = channelsForProducts.stream()
+                                    .filter(c -> c.isBaseChannel()).findFirst();
```

Comment:
We have the bad situation that we have products with multiple base channels (RES6).
Check if we can find the architecture.

Marcello just added a new function to this class called: `filterCompatibleChannelsForServerArch()` which might be helpful.

Reply:
Btw: he changed the part directly below your changes. Please rebase.

Reply:
That's what I did not know at all and I asked you, great you pointed out about the RES6 case.
And thanks for the hints. I will check it out and rebase. :+1: